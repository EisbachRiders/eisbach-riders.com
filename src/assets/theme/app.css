html {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Raleway', sans-serif;
  background-color: #ffffff;
  overflow-x: hidden;
}

a {
  text-decoration: none;
}
