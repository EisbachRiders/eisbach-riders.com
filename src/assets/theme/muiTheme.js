import { createMuiTheme } from '@material-ui/core/styles'
import './App.css'
require('typeface-open-sans')
require('typeface-contrail-one')

const status = {
  white: '#ffffff',
  offWhite: '#F7F5F5',
  black: '#0b0c10',
  blue: '#1f2833',
  grey: '#c5c6c7',
  greyDk: '#404040',
  teal: '#66fcf1',
  tealDk: '#45a29e',
  green: '#66fca6',
  yellow: '#fcf166',
  red: '#e25b65',
}

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: status.teal,
      main: status.teal,
      dark: status.tealDk,
    },
    secondary: {
      light: status.blue,
      main: status.blue,
      dark: status.blue,
    },
    error: { main: status.red },
  },
  status,
  typography: {
    fontFamily: "'Open Sans', sans-serif",
    fontSize: 14,
  },
  overrides: {
    MuiTypography: {
      colorPrimary: {
        color: status.greyDk,
      },
      colorSecondary: {
        color: status.grey,
      },
      colorTextSecondary: {
        color: status.white,
      },
    },
  },
})

export default theme
