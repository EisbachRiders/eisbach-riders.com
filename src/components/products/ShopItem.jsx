import React from "react"
import { Link } from "gatsby-theme-material-ui"
import { makeStyles } from "@material-ui/core/styles"
import Typography from "@material-ui/core/Typography"
import Colors from "./Colors"

const useStyles = makeStyles((theme) => ({
  root: {
    flexBasis: "100%",
    alignItems: "center",
    display: "flex",
    flexDirection: "column",
    marginBottom: 30,
    [theme.breakpoints.up("sm")]: {
      flexBasis: "50%",
    },
    [theme.breakpoints.up("md")]: {
      flexBasis: "22%",
    },
  },
  img: {
    width: "100%",
  },
  descriptionContainer: {
    height: "100%",
    width: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
  },
  name: {
    marginBottom: 5,
    fontSize: 14,
    fontWeight: 300,
    textTransform: "capitalize",
    textAlign: "center",
    width: "100%",
  },
  priceContainer: {
    display: "flex",
    justifyContent: "space-between",
  },
  price: {
    fontSize: 14,
    marginBottom: 5,
  },
  red: {
    color: theme.color.red,
    textTransform: "capitalize",
  },
}))

function ShopItem({ product }) {
  const classes = useStyles()
  let colors =
    product === undefined
      ? []
      : product.paColors.edges.map((elem) => elem.node.name)

  return (
    <div className={classes.root}>
      {product && (
        <>
          <Link to={`/products/${product.slug}`}>
            <img
              src={product.image.sourceUrl}
              className={classes.img}
              alt={product.image.slug || product.name}
            />
          </Link>
          <div className={classes.descriptionContainer}>
            <Typography className={classes.name}>{product.name}</Typography>
            {colors.length > 1 && <Colors colors={colors} variant="small" />}
          </div>
        </>
      )}
    </div>
  )
}

export default ShopItem
