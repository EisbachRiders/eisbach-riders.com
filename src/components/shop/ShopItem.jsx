import React from 'react'
import PropTypes from 'prop-types'
import Link from '../Link'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import { connect } from 'react-redux'
import classnames from 'classnames'
import Colors from './Colors'

const mapDispatchToProps = dispatch => {
  return { changeProduct: () => dispatch({ type: `Product` }) }
}

const useStyles = makeStyles(theme => ({
  root: {
    flexBasis: '50%',
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    marginBottom: 30,
    [theme.breakpoints.up('sm')]: {
      flexBasis: '22%',
    },
  },
  img: {
    width: '80%',
    [theme.breakpoints.up('sm')]: {
      width: '100%',
    },
  },
  descriptionContainer: {
    height: '100%',
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  name: {
    marginBottom: 5,
    fontSize: 14,
    fontWeight: 300,
    textTransform: 'capital',
    color: theme.status.greyMed,
    width: '100%',
  },
  priceContainer: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  price: {
    fontSize: 14,
    // fontWeight: 300,
    marginBottom: 5,
  },
  red: {
    color: theme.status.red,
    textTransform: 'capitalize',
  },
}))

function ShopItem({ product }) {
  const classes = useStyles()
  const { t } = useTranslation()

  const placeholder = !product.images
  const colors =
    product.attributes &&
    product.attributes.length !== 0 &&
    product.attributes.find(e => e.name === 'Color')
      ? product.attributes.find(e => e.name === 'Color').options
      : []

  return (
    <div className={classes.root}>
      {!placeholder && (
        <>
          <Link to={'/product/'}>
            <img
              src={product.images[0].src}
              className={classes.img}
              alt={product.name}
            />
          </Link>
          <div className={classes.descriptionContainer}>
            <div>
              <div className={classes.priceContainer}>
                <Typography className={classes.price}>{`â‚¬${
                  product.price
                }`}</Typography>
                {!product.in_stock && (
                  <Typography
                    className={classnames(classes.price, classes.red)}
                  >
                    {t('products.outOfStock')}
                  </Typography>
                )}
              </div>
              <Typography className={classes.name}>{product.name}</Typography>
            </div>
            <Colors colors={colors} />
          </div>
        </>
      )}
    </div>
  )
}

ShopItem.propTypes = {
  product: PropTypes.object.isRequired,
}

export default connect(mapDispatchToProps)(ShopItem)
