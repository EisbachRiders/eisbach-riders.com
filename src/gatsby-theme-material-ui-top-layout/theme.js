import { createMuiTheme } from "@material-ui/core/styles"
import "../app.css"
import "../../static/fonts/fonts.css"

// fin: "#00d7a2",
// finDk: "#00be8f",
// finLt: "#e2f9ef",
// greyLt2: "#eeeff0",
// greyLt: "#d5d9db",
// grey: "#818C93",
// greyBlue: "#f3f8fc",
// greenLt1: "#6dff97",
// greenLt2: "#00cf3c",
// cream: "#f7f5ee",

const color = {
  // for shop colors
  blue: "#004AB5",
  green: "#00942b",
  yellow: "#fce566",
  orange: "#ff8000",
  pink: "#ff0080",
  red: "#FF0012",
  redLt: "#ff4e5b",
  // theme colors
  white: "#ffffff",
  black: "#000000",
  almostBlack: "#363738", // typography
  textLight: "#838588",
  gray: "#ebedee",
  primaryBlue: "#0062f0",
  primaryBlueLt: "#3f8eff",
  gold: "#E6C55E",
  goldLt: "#f0dda0",
}

const theme = createMuiTheme({
  palette: {
    type: "light",
    primary: {
      light: color.goldLt,
      main: color.gold,
      dark: color.goldLt,
    },
    secondary: {
      light: color.primaryBlueLt,
      main: color.primaryBlue,
      dark: color.primaryBlueLt,
    },
    error: { main: color.red },
  },
  color,
  typography: {
    fontFamily: "'Raleway', sans-serif",
    h1: {
      letterSpacing: 2,
    },
    h2: {
      letterSpacing: 2,
    },
    h3: {
      letterSpacing: 2,
    },
    h4: {
      letterSpacing: 2,
    },
    h5: {
      letterSpacing: 2,
    },
    body1: {
      letterSpacing: 1.5,
      fontSize: 14,
    },
    body2: {
      letterSpacing: 1.5,
      fontSize: 12,
    },
    button: {
      fontFamily: "'Raleway', sans-serif",
      letterSpacing: 3,
    },
  },
  overrides: {
    MuiButton: {
      root: {
        padding: "9px 27px",
        borderRadius: 0,
      },
      contained: {
        boxShadow: "none",
      },
    },
  },
})

export default theme
