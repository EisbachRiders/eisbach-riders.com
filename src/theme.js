import { createTheme } from "@material-ui/core/styles"
import "./css/app.css"

// fin: "#00d7a2",
// finDk: "#00be8f",
// finLt: "#e2f9ef",
// grayLt2: "#eeeff0",
// grayLt: "#d5d9db",
// gray: "#818C93",
// grayBlue: "#f3f8fc",
// greenLt1: "#6dff97",
// greenLt2: "#00cf3c",
// cream: "#f7f5ee",

const color = {
  // for shop colors
  blue: "#004AB5",
  green: "#00942b",
  yellow: "#fce566",
  orange: "#ff8000",
  pink: "#ff0080",
  red: "#FF0012",
  redLt: "#ff4e5b",
  seaGlass: "#5F5850",
  // theme colors
  white: "#ffffff",
  black: "#0B0C10",
  blackLt: "#1F2833",
  gray: "#C5C6C7",
  // teal: "#66FCF1",
  teal: "#00d7a2",
  tealDk: "#45A29E",
  test: "#F7F680",
  cream: "#F1F0EB",
  grayLt: "#f7f7f9",
  blueGray: "#cbd3cf",
  steel: "#505D72",
}

const theme = createTheme({
  palette: {
    type: "light",
    primary: {
      light: color.tealDk,
      main: color.teal,
      dark: color.tealDk,
    },
    secondary: {
      light: "#9d9fa0",
      main: color.gray,
      dark: "#9d9fa0",
    },
    error: { main: color.red },
  },
  color,
  typography: {
    fontFamily: ["Raleway", "sans-serif"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          padding: "9px 27px",
          borderRadius: 0,
        },
        contained: {
          boxShadow: "none",
        },
      },
    },
  },
})

export default theme
