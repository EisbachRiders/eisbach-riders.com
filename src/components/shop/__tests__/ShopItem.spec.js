import React from 'react'
import { render } from 'react-testing-library'
import ShopItem from '../ShopItem'
import { ThemeProvider } from '@material-ui/styles'
import theme from '../../../assets/theme/muiTheme'

describe('ShopItem', () => {
  it('renders correctly', () => {
    const mockProps = {
      product: {
        images: [],
        more: { en: [] },
      },
      t: () => '',
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <ShopItem {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
