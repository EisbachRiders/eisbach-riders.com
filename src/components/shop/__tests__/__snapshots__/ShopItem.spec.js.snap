// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopItem renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-o7yfbk"
      >
        <img
          alt="product test"
          class="Hook-img-16x9eyn"
        />
        <div
          class="Hook-descriptionContainer-12gz4p"
        >
          <p
            class="MuiTypography-root-1 MuiTypography-body2-2"
          />
          <div
            class="Hook-fullWidth-115baqa"
          >
            <div
              class="Hook-priceContainer-10ca940"
            >
              <h6
                class="MuiTypography-root-1 MuiTypography-h6-11 Hook-price-169xtyo"
              >
                €undefined
              </h6>
              <div />
            </div>
          </div>
          <button
            class="MuiButtonBase-root-48 MuiButton-root-31 MuiButton-outlined-36 MuiButton-outlinedSecondary-38 Hook-button-1ecjfgm Hook-buttonFull-twp37v"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-32"
            >
              features
            </span>
            <span
              class="MuiTouchRipple-root-68"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-o7yfbk"
    >
      <img
        alt="product test"
        class="Hook-img-16x9eyn"
      />
      <div
        class="Hook-descriptionContainer-12gz4p"
      >
        <p
          class="MuiTypography-root-1 MuiTypography-body2-2"
        />
        <div
          class="Hook-fullWidth-115baqa"
        >
          <div
            class="Hook-priceContainer-10ca940"
          >
            <h6
              class="MuiTypography-root-1 MuiTypography-h6-11 Hook-price-169xtyo"
            >
              €undefined
            </h6>
            <div />
          </div>
        </div>
        <button
          class="MuiButtonBase-root-48 MuiButton-root-31 MuiButton-outlined-36 MuiButton-outlinedSecondary-38 Hook-button-1ecjfgm Hook-buttonFull-twp37v"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-32"
          >
            features
          </span>
          <span
            class="MuiTouchRipple-root-68"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
